remote-api/src/main/java/org/alfresco/rest/api/impl/CategoriesImpl.java

```diff
@@ -0,0 +1,108 @@
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.rest.api.impl;
+
+import static org.alfresco.rest.api.Nodes.PATH_ROOT;
+
+import java.util.List;
+import java.util.Set;
+
+import org.alfresco.model.ContentModel;
+import org.alfresco.rest.api.Categories;
+import org.alfresco.rest.api.Nodes;
+import org.alfresco.rest.api.model.Category;
+import org.alfresco.rest.api.model.Node;
+import org.alfresco.rest.framework.core.exceptions.EntityNotFoundException;
+import org.alfresco.rest.framework.core.exceptions.InvalidArgumentException;
+import org.alfresco.rest.framework.resource.parameters.Parameters;
+import org.alfresco.service.Experimental;
+import org.alfresco.service.cmr.repository.ChildAssociationRef;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.NodeService;
+import org.alfresco.service.cmr.repository.StoreRef;
+import org.alfresco.service.namespace.RegexQNamePattern;
+import org.apache.commons.collections.CollectionUtils;
+
+@Experimental
+public class CategoriesImpl implements Categories
+{
+    private final Nodes nodes;
+    private final NodeService nodeService;
+
+    public CategoriesImpl(Nodes nodes, NodeService nodeService)
+    {
+        this.nodes = nodes;
+        this.nodeService = nodeService;
+    }
+
+    @Override
+    public Category getCategoryById(String id, Parameters params)
+    {
+        final NodeRef nodeRef = PATH_ROOT.equals(id) ? getRooCategoryNodeRef() : nodes.validateNode(id);
+        final boolean isCategory = nodes.isSubClass(nodeRef, ContentModel.TYPE_CATEGORY, false);
+        if (!isCategory)
+        {
+            throw new InvalidArgumentException("Node id does not refer to a category", new String[]{id});
+        }
+        final Node categoryNode = nodes.getNode(nodeRef.getId());
+        final Category category = new Category();
+        category.setId(nodeRef.getId());
+        category.setName(categoryNode.getName());
+        category.setParentId(getParentId(nodeRef));
+        final boolean hasChildren = CollectionUtils
+                .isNotEmpty(nodeService.getChildAssocs(nodeRef, RegexQNamePattern.MATCH_ALL, RegexQNamePattern.MATCH_ALL, false));
+        category.setHasChildren(hasChildren);
+
+        return category;
+    }
+
+    @Override
+    public NodeRef getRooCategoryNodeRef()
```

Comment:
Typo - missing t: getRooCategoryNodeRef -> getRootCategoryNodeRef

Reply:
Thanks, I moved that method out to `CategoryService` interface but fixed the TYPO too.